src/core/conversion/dataIndexer.ts

```diff
@@ -0,0 +1,183 @@
+import { IQueryVisualizationResponse } from "../types";
+import { EVariableType, IVariableMeta, IVariableValueMeta, TMultiLanguageString } from "../types/queryVisualizationResponse";
+import { TVariableSelections } from "../types/variableSelections";
+import { IDataCell, IDataSeries } from "../types/view";
+import { cartesianProduct } from "./utilityFunctions";
+import { isEqual } from "lodash";
+
+export class DataIndexer {
+    public dataIndex: number;
+    public dataLength: number;
+
+    private responseObj: IQueryVisualizationResponse;
+    private coordinates: number[][];
+    private lastIndices: number[];
+    private indices: number[];
+    private variableOrder: number[];
+    private lastCoordinateIndex: number;
+    private reverseCumulativeProducts: number[];
+    private contentVariableIndex: number;
+    private timeVariableIndex: number;
+    private targetMap: IVariableMeta[];
```

Comment:
targetMap could be renamed selectedViewMeta or somethig simillar more descriptive.